File: AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/Base Classes/ALBaseAdViewController.m
//
//  ALBaseAdViewController.m
//  DemoApp-ObjC
//
//  Created by Harry Arakkal on 10/9/19.
//  Copyright © 2019 AppLovin Corporation. All rights reserved.
//

#import "ALBaseAdViewController.h"

@interface ALBaseAdViewController ()<UITableViewDataSource>
@property (nonatomic, strong) NSMutableArray<NSString *> *callbacks;

@end

@implementation ALBaseAdViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.callbacks = [NSMutableArray array];
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear: animated];
    [self.navigationController setToolbarHidden: self.hidesBottomBarWhenPushed animated: YES];
}

- (void)viewWillDisappear:(BOOL)animated
{
    [self.navigationController setToolbarHidden: YES];
    [super viewWillDisappear: animated];
}

- (void)logCallback:(const char *)name
{
    [self.callbacks addObject: [NSString stringWithCString: name encoding: NSUTF8StringEncoding]];
    
    NSArray<NSIndexPath *> *lastIndexPath = @[[NSIndexPath indexPathForRow: self.callbacks.count - 1 inSection: 0]];
    [self.callbackTableView insertRowsAtIndexPaths: lastIndexPath withRowAnimation: UITableViewRowAnimationAutomatic];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier: @"callbackCell" forIndexPath: indexPath];
    cell.textLabel.text = self.callbacks[indexPath.row];
    
    return cell;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.callbacks.count;
}

@end

File: AppLovin MAX Demo App - Swift/AppLovin MAX Demo App - Swift/Base Classes/ALBaseAdViewController.swift
//
//  ALBaseAdViewController.swift
//  DemoApp-Swift
//
//  Created by Harry Arakkal on 10/7/19.
//  Copyright © 2019 AppLovin. All rights reserved.
//

import AppLovinSDK
import UIKit

class ALBaseAdViewController: UIViewController
{
    @IBOutlet weak var callbackTableView: UITableView!
    
    private var callbacks: [String] = []
    
    override func viewDidAppear(_ animated: Bool)
    {
        super.viewDidAppear(animated)
    self.navigationController?.setToolbarHidden(self.hidesBottomBarWhenPushed, animated: true)
    }

    override func viewWillDisappear(_ animated: Bool)
    {
        self.navigationController?.setToolbarHidden(true, animated: false)
        super.viewWillDisappear(animated)
    }

    internal func logCallback(functionName: String = #function)
    {
        callbacks.append(functionName)
        callbackTableView.insertRows(at: [IndexPath(row: callbacks.count - 1, section: 0)], with: .automatic)
    }
}

extension ALBaseAdViewController: UITableViewDataSource
{
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
    {
        let cell = tableView.dequeueReusableCell(withIdentifier: "callbackCell", for: indexPath)
        cell.textLabel?.text = callbacks[indexPath.row]

        return cell
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
        return callbacks.count
    }
}
